brainio/lookup.py

```diff
@@ -5,66 +5,68 @@
 import entrypoints
 import numpy as np
 import pandas as pd
+from brainio.catalogs import Catalog, SOURCE_CATALOG
 
-LOOKUP_SOURCE = "lookup_source"
 ENTRYPOINT = "brainio_lookups"
 TYPE_ASSEMBLY = 'assembly'
 TYPE_STIMULUS_SET = 'stimulus_set'
-CATALOG_PATH_KEY = "catalog_path"
-_catalogs = {}
+_catalogs = None
 _concat_catalogs = None
 
 _logger = logging.getLogger(__name__)
 
 
-def list_catalogs():
+def list_installed_catalogs():
     return list(entrypoints.get_group_named(ENTRYPOINT).keys())
 
 
-def load_lookup(name, entry_point):
-    df = entry_point.load()()
-    df[LOOKUP_SOURCE] = name
-    return df
+def _load_catalog(identifier, entry_point):
+    catalog = entry_point.load()()
+    assert isinstance(catalog, Catalog)
+    assert catalog.identifier == identifier
+    return catalog
 
 
-def get_lookups():
-    lookups = entrypoints.get_group_named(ENTRYPOINT)
-    dfs = {}
-    for k, v in lookups.items():
-        df = load_lookup(k, v)
-        dfs[k] = df
-    return dfs
+def _load_installed_catalogs():
+    installed_catalogs = entrypoints.get_group_named(ENTRYPOINT)
+    catalogs = {}
+    for k, v in installed_catalogs.items():
+        catalog = _load_catalog(k, v)
+        catalogs[k] = catalog
+    return catalogs
 
 
 def get_catalogs():
     global _catalogs
-    if not _catalogs:
-        _logger.debug(f"Loading lookup from entrypoints")
-        print(f"Loading lookup from entrypoints")
-        _catalogs = get_lookups()
+    if _catalogs is None:
+        _logger.debug(f"Loading catalog from entrypoints")
+        print(f"Loading catalog from entrypoints")
+        _catalogs = _load_installed_catalogs()
     return _catalogs
 
 
-def data():
+def combined_catalog():
     global _concat_catalogs
     if _concat_catalogs is None:
```

Comment:
is this an expensive operation? If not, I would remove the global variable cache.

Reply:
If you mean  _concat_catalogs, agreed.